crates/core/src/node/console.rs

```diff
@@ -0,0 +1,90 @@
+use alloy::primitives::{hex, I256, U256};
+use alloy::sol;
+use derive_more::Display;
+use itertools::Itertools;
+
+sol! {
+#[sol(abi)]
+#[derive(Display)]
+interface Console {
+    #[display("{val}")]
+    event log(string val);
+
+    #[display("{}", hex::encode_prefixed(val))]
+    event logs(bytes val);
+
+    #[display("{val}")]
+    event log_address(address val);
+
+    #[display("{val}")]
+    event log_bytes32(bytes32 val);
+
+    #[display("{val}")]
+    event log_int(int val);
+
+    #[display("{val}")]
+    event log_uint(uint val);
+
+    #[display("{}", hex::encode_prefixed(val))]
+    event log_bytes(bytes val);
+
+    #[display("{val}")]
+    event log_string(string val);
+
+    #[display("[{}]", val.iter().format(", "))]
+    event log_array(uint256[] val);
+
+    #[display("[{}]", val.iter().format(", "))]
+    event log_array(int256[] val);
+
+    #[display("[{}]", val.iter().format(", "))]
+    event log_array(address[] val);
+
+    #[display("{key}: {val}")]
+    event log_named_address(string key, address val);
+
+    #[display("{key}: {val}")]
+    event log_named_bytes32(string key, bytes32 val);
+
+    #[display("{key}: {}", format_units_int(val, decimals))]
+    event log_named_decimal_int(string key, int val, uint decimals);
+
+    #[display("{key}: {}", format_units_uint(val, decimals))]
+    event log_named_decimal_uint(string key, uint val, uint decimals);
+
+    #[display("{key}: {val}")]
+    event log_named_int(string key, int val);
+
+    #[display("{key}: {val}")]
+    event log_named_uint(string key, uint val);
+
+    #[display("{key}: {}", hex::encode_prefixed(val))]
+    event log_named_bytes(string key, bytes val);
+
+    #[display("{key}: {val}")]
+    event log_named_string(string key, string val);
+
+    #[display("{key}: [{}]", val.iter().format(", "))]
+    event log_named_array(string key, uint256[] val);
+
+    #[display("{key}: [{}]", val.iter().format(", "))]
+    event log_named_array(string key, int256[] val);
+
+    #[display("{key}: [{}]", val.iter().format(", "))]
+    event log_named_array(string key, address[] val);
+}
+}
```

Comment:
A couple of nits:
* Rename the file to `forge_console.rs` to disambiguate
* IMO `core/node` is not the best place for these. Maybe let's create a new crate that holds custom support cheat contracts?
* Wouldn't it be better to have matching approaches here and in `hardhat_console.rs`? i.e. let's use an ABI file instead of listing each event manually which seems very error-prone

Reply:
The reason they are listed manually is because `ConsoleFmt` is not implemented for specific types and when we decode the logs we need them pretty printed. This is the same reason why upstream has it the [same](https://github.com/foundry-rs/foundry/pull/8910/files#diff-53f21ddcb80390cd3bea462f6c49e9f9bdc9d647745280d34f748f09b3ff756d) for right now.